src/GenericCyclotomics.jl

```diff
@@ -365,10 +365,15 @@ end
 
 mul!(x::GenericCyclo, y::GenericCyclo, z::GenericCyclo) = y*z
 
-add!(x::GenericCyclo, y::GenericCyclo, z::GenericCyclo) = y+z
-
-function addeq!(x::GenericCyclo, y::GenericCyclo)
-	mergewith!(+, x.f, y.f)
+function add!(x::GenericCyclo, y::GenericCyclo, z::GenericCyclo)
+	if x == z
+		y,z=z,y
```

Comment:
What if all three are equal?

Reply:
In this case the swap is unnecessary and useless but not harmful.

Reply:
But `==` should actually be `===` I think.

Reply:
OK, so `mergewith` works correctly if both arguments are the same?

Reply:
Yes, I was worried too.
```
julia> a=Dict(["a" => 1, "b" => 2])
Dict{String, Int64} with 2 entries:
  "b" => 2
  "a" => 1

julia> mergewith!(+, a, a)
Dict{String, Int64} with 2 entries:
  "b" => 4
  "a" => 2


```